File: CineMatic.Model/DTO/RezervacijaFilmDTO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CineMatic.Model.DTO
{
    public class RezervacijaFilmDTO
    {
        public int RezervacijaId { get; set; }
        public DateTime DatumRezervacije { get; set; }
        public string? NacinPlacanja { get; set; }
        public int ProjekcijaId { get; set; }
        public DateTime DatumProjekcije { get; set; }
        public List<int> SjedistaIds { get; set; } = new List<int>();
        public List<int> HranaPiceIds { get; set; } = new List<int>();

        public int FilmId { get; set; }
        public string? NazivFilma { get; set; }
        public int? TrajanjeFilma { get; set; }
        public string? Opis { get; set; }
        public string? FilmaSlikaBase64 { get; set; }
    }
}

File: UI/cinematic_mobile/lib/models/rezervacija_film_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija_film_dto.g.dart';

@JsonSerializable()
class RezervacijaFilmDTO {
  int? rezervacijaId;
  DateTime? datumRezervacije;
  String? nacinPlacanja;
  int? projekcijaId;
  DateTime? datumProjekcije;
  List<int>? sjedistaIds;
  List<int>? hranaPiceIds;
  List<int>? kolicine;
  double? ukupnaCijena;
  String? qrCodeBase64;
  int? filmId;
  String? nazivFilma;
  int? trajanjeFilma;
  String? opis;
  String? filmaSlikaBase64;

  RezervacijaFilmDTO({
    this.rezervacijaId,
    this.datumRezervacije,
    this.nacinPlacanja,
    this.projekcijaId,
    this.datumProjekcije,
    this.sjedistaIds,
    this.hranaPiceIds,
    this.filmId,
    this.nazivFilma,
    this.trajanjeFilma,
    this.opis,
    this.filmaSlikaBase64,
  });

  factory RezervacijaFilmDTO.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFilmDTOFromJson(json);

  Map<String, dynamic> toJson() => _$RezervacijaFilmDTOToJson(this);
}